cdap-docs/reference-manual/source/http-restful-api/http-restful-api-v3/metrics.rst

```diff
@@ -98,114 +373,162 @@ The general form of a metrics request is::
+    &metric=system.process.events.processed&start=now-5s&count=5
 
-  GET <base-url>/metrics/system/apps/CountRandom/
-    flows/process.events.processed?start=now-5s&count=5
-  GET <base-url>/metrics/system/apps/CountRandom/
-    process.events.processed?start=now-5s&count=5
-  GET <base-url>/metrics/system/process.events?start=now-5s&count=5
+  POST <base-url>/metrics/query?context=namespace.default
+    &metric=system.process.events.processed&start=now-5s&count=5
 
 To request user-defined metrics instead of system metrics, specify ``user`` instead of ``cdap`` in the URL
 and specify the user-defined metric at the end of the request.
 
 For example, to request a user-defined metric for the *HelloWorld* Application's *WhoFlow* Flow::
 
-  GET <base-url>/metrics/user/apps/HelloWorld/flows/
-    WhoFlow/flowlets/saver/names.bytes?aggregate=true
+  POST <base-url>/metrics/query?context=namespace.default.app.HelloWorld.flow.WhoFlow.flowlet.saver
+    &metric=user.names.bytes&aggregate=true
 
-To retrieve multiple metrics at once, instead of a GET, issue an HTTP POST, with a JSON list as the request body that enumerates the name and attributes for each metrics. For example::
+Retrieving multiple metrics at once, by issuing an HTTP POST request with a JSON list as
+the request body that enumerates the name and attributes for each metric, is currently not
+supported in this API. Instead, use the :ref:`v2 API
+<http-restful-api-v2-metrics-multiple>`. It will be supported in a future release.
 
-  POST <base-url>/metrics
 
-with the arguments as a JSON string in the body::
+.. _http-restful-api-metrics-tag-list:
 
-  Content-Type: application/json
-  [ "/system/collect.events?aggregate=true",
-  "/system/apps/HelloWorld/process.events.processed?start=1380323712&count=6000" ]
+Tag List
+--------
 
-If the context of the requested metric or metric itself doesn't exist the system returns status 200 (OK) with JSON formed as per above description and with values being zeroes.
+In a query, the optional ``groupBy``​parameter defines a list of tags whose values are
+used to build multiple timeseries. All data points that have the same values in tags
+specified in the ``groupBy`` parameter will form a single timeseries. You define
+multiple tags for grouping by providing a comma-separated list.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Tag List
+     - Description
+   * - ``groupBy=app``
+     - Retrieves the time series for each application. 
+       For example: ``now-5d-12h`` is 5 days and 12 hours ago.
+   * - ``groupBy=app,flow``
+     - Retrieves a time series for each app and flow combination
+
+.. _http-restful-api-metrics-time-range:
 
 Time Range
 ----------
-The time range of a metric query can be specified in various ways:
+The time range of a metric query can be specified in various ways: either
+``aggregate=true`` to retrieve the total aggregated since the Application was deployed, 
```

Comment:
don't put "since the application was deployed" - you can query not only for app metrics, right?